### ISSUE TITLE
`function --argument-names` does not error for read-only variables

### ISSUE THREAD

**Issue by kopischke** [fish-shell/fish-shell#3073]
When defining a named argument to a function with the same name as a lower-case Fish special variable, the variable is not shadowed by the corresponding `$argv` value but assigned to the argument instead.
### Reproduction Steps:
1. `function testver --argument-names version; echo $version; end`
2. `testver 1.2`
### Expected behavior:

Call outputs “1.2”
### Observed behavior:

Call outputs “2.3”
### Additional information:
- Reproducible with other Fish specific special variables like `$status`
- Not reproducible with common exported shell variables like `$HOME`

---
- Fish version: 2.3, Homebrew installed via `brew install fish --devel`
- Operating system: OS X 10.11.4
- Terminal or terminal emulator: Terminal.app

**Comment by faho** [comment]
I can reproduce this.

For `$status` and such, we probably want to error out when you try to use them here - like we already do when you try to write to them.

For `$version`, that should be removed.

Can you find any variable causing an issue that is not one of status, history, version, `$_`, LINES, COLUMNS or PWD, FISH_VERSION (and possibly SHLVL)? That's our full list of read-only variables, so if the issue is only with them, we should just add an error message.

**Comment by kopischke** [comment]
> For `$status` and such, we probably want to error out when you try to use them here - like we already do when you try to write to them.

Hmmm … AFAICS, declaring a named argument means the argument is in function local scope, shadowing any outer scope variable, i.e.

```fish
set -gx MYVAR foo
function testvar --argument-names MYVAR
    echo $MYVAR
end
testvar bar
echo $MYVAR
```

echoes “bar”, then “foo”. I would expect special variables to be shadowed the same way, not to get special treatment (or rather: a different form of special treatment than the one they get now).

**Comment by faho** [comment]
@kopischke: These special variables can't be shadowed. Which is also why a generic name like "$version" is so bad.

E.g. $status should _not_ be shadowed by anything else - it would render large parts of your shell unusable.

**Comment by kopischke** [comment]
> E.g. $status should not be shadowed by anything else - it would render large parts of your shell unusable.

True, that; I was mentally stuck on `version`. Disregard my previous comment.

**Comment by faho** [comment]
Okay, $version has been removed (95635a5982ec8ab234d925a4bb5025328a706fa3).

The current behavior is still that you can _try_ to shadow these variables, it just won't work.
